apps/src/javalab/JavabuilderConnection.js

```diff
@@ -188,12 +195,24 @@ export default class JavabuilderConnection {
     // happen if our token was somehow valid for the initial Javabuilder HTTP request and
     // then became invalid when establishing the WebSocket connection.
     this.sendMessage(WebSocketMessageType.CONNECTED);
+    // If we don't receive a message back within 10 seconds, Javabuilder may be at maximum capacity and
+    // the request will be queued to execute when an instance is available. Notify the user that this may
+    // be the case.
+    setTimeout(() => {
```

Comment:
Should we also save the timeout ID and clear it once we receive a message?

Reply:
I'm not sure. It would save us from doing the check after 10 seconds, but we'd either have to assume the first message will always be `COMPILING...` (I think that's the case, but there could be an error case I'm not thinking of), still do the check for whether we've seen a message and only clear once, or clear on every message, somewhat redundantly. I think clearing on every message has a little risk since the timeout id might get reused by a future timeout. I'm inclined to leave as-is since it doesn't really simplify things, but am curious if there's a big optimization to be had by clearing it.

Reply:
Yep makes sense, agreed that there's minimal risk and I don't think we'd gain a ton from optimization